packages/guui/dropdown/index.js

```diff
@@ -228,20 +241,22 @@ export default class Dropdown extends Component<
 
                 <ul
                     id={dropdownID}
-                    className={this.state.isExpanded ? 'expanded' : ''}
+                    className={this.state.isExpanded ? ulExpanded : ul}
                 >
-                    {links.map(link => (
-                        <li key={link.title}>
+                    {links.map((l, index) => (
+                        <li key={l.title}>
                             <a
-                                href={link.url}
-                                className={link.isActive ? 'active' : ''}
+                                href={l.url}
+                                className={`${l.isActive ? linkActive : link} ${
+                                    index === 0 ? linkFirst : ''
+                                }`}
```

Comment:
The bit above is a bit fiddly and currently breaks our 'one class per element' rule. Any thoughts on how to improve this welcome!

Reply:
I'm wondering whether the single className rule is a premature optimisation that shifts the inefficiencies elsewhere (duplicating style rules across multiple classes). I also think it could lead to fiddly issues like this where we have to write more JS logic to pick the correct single classname.

Reply:
Hey, yeah I'm not sure. It's not too hard to rewrite it as a function:

    const linkStyle = (isActive, isFirst) => {
        if (isActive && isFirst) {
            return activeAndFIrst
        } else if (isFirst) {
            return firstLink
        } else (isActive) {
            return activeLink
        }

        return normalLink
    }

Performance-wise emotion seems to be pretty clear that generating styles dynamically is bad - so the idea is to pre-declare them like above.

Reply:
Hey, my preference right now is multiple class names, eg:

```JS
<Link className={`baseLinkStyles ${isActive ? 'activeLinkStyles': ''} ${isFirst ? 'firstLinkStyles': ''}`}>Click me</Link>
```
There are libs like [classNames](https://www.npmjs.com/package/classnames) that can be used for conditionally creating classNames too.

I'd say the benefits of this are:
- Less duplication of style rules within the source code, if you want to change the  "base" rule of link you update it once.
- Less duplication of style rules within generated CSS
- More css-like to combine styles this way

Reply:
After some discussion we've concluded that [`cx`](https://github.com/emotion-js/emotion/blob/master/docs/cx.md), the Emotion-specific implementation of `classNames` generates the most performant CSS in this case.

```js
<Link className={
  cx({
    [link]: true,
    [linkActive]: isActive, 
    [linkFirst] isFirst,
  })
>Click me</Link>
```